test/Escrow.js

```diff
@@ -310,6 +314,34 @@ contract('Escrow', (accounts) => {
       }
     });
 
+    it('optionally writes out internal url & hash', async () => {
+      try {
+        const toAddress = [accounts[0]];
+        const initialStatus = await Escrow.getStatus.call();
+        assert.equal(initialStatus, 0);
+
+        // Transer funds to escrow
+        await HMT.transfer(Escrow.address, 200, { from: accounts[0] });
+
+        // Setup escrow
+        await Escrow.setup(reputationOracle, recordingOracle, 10, 10, recOIpns, repOIpns, url, hash, { from: accounts[0] });
+        const setupStatus = await Escrow.getStatus.call();
+        assert.equal(setupStatus, 1);
+
+        const amountToPay = [4];
+        assert(await Escrow.getReputationOracleResultsUrl.call() === '' && await Escrow.getReputationOracleResultsHash.call() === '',
+              'hash and url should be empty');
+        await Escrow.bulkPayOut(toAddress, amountToPay, url, hash, '000', { from: reputationOracle });
+        assert(await Escrow.getReputationOracleResultsUrl.call() === url && await Escrow.getReputationOracleResultsHash.call() === hash,
+              'either hash or url not set');
+        await Escrow.bulkPayOut(toAddress, amountToPay, '', '', '000', { from: reputationOracle });
+        assert(await Escrow.getReputationOracleResultsUrl.call() === url && await Escrow.getReputationOracleResultsHash.call() === hash,
+                    'hash or url changed when it should not have');
```

Comment:
where's the test to make sure ipfs updates?

Reply:
622 and 629 show the updating
https://github.com/hCaptcha/hmt-escrow/blob/bc5a2b230c5f0e6cbf384aabc4185dab0132e25a/hmt_escrow/job.py#L629

the js file is just setting the ipns link. so ipfs updates aren't touched on the blockchain

Reply:
@posix4e , this was just added in hmt_escrow/test_ipns.py 
ill mark resolved